mdtraj/geometry/hbond.py

```diff
@@ -141,9 +148,14 @@ def get_donors(e0, e1):
 
         return indices
 
-    nh_donors = get_donors('N', 'H')
-    oh_donors = get_donors('O', 'H')
-    xh_donors = np.array(nh_donors + oh_donors)
+    nh_donors = get_donors('N', 'H', proposed_donor_indices=proposed_donor_indices)
+    oh_donors = get_donors('O', 'H', proposed_donor_indices=proposed_donor_indices)
+    if nh_donors != [] and oh_donors != []:
```

Comment:
Similarly, the more robust thing to do here is check `if len(x) > 0` (see line 160 for an example)